Validation/RecoMuon/plugins/NewMuonTrackValidatorBase.h

```diff
@@ -0,0 +1,285 @@
+#ifndef NewMuonTrackValidatorBase_h
+#define NewMuonTrackValidatorBase_h
+
+/** \class NewMuonTrackValidatorBase
+* Base class for analyzers that produces histograms to validate Muon Track Reconstruction performances
+*
+*/
+
+#include <memory>
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticleFwd.h"
+#include "SimDataFormats/PileupSummaryInfo/interface/PileupSummaryInfo.h"
+
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "SimTracker/Common/interface/TrackingParticleSelector.h"
+#include "CommonTools/RecoAlgos/interface/CosmicTrackingParticleSelector.h"
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <TH1F.h>
+#include <TH2F.h>
+
+
+class DQMStore;
+class NewMuonTrackValidatorBase {
+ public:
+  /// Constructor
+  NewMuonTrackValidatorBase(const edm::ParameterSet& pset, edm::ConsumesCollector iC) : NewMuonTrackValidatorBase(pset)
+    {
+      bsSrc_Token = iC.consumes<reco::BeamSpot>(bsSrc);
+      tp_effic_Token = iC.consumes<TrackingParticleCollection>(label_tp_effic);
+      tp_fake_Token = iC.consumes<TrackingParticleCollection>(label_tp_fake);
+      pileupinfo_Token = iC.consumes<std::vector<PileupSummaryInfo> >(label_pileupinfo);
+      for (unsigned int www=0;www<label.size();www++){
+	track_Collection_Token[www] = iC.consumes<edm::View<reco::Track> >(label[www]);
+      }
+    }
+
+  NewMuonTrackValidatorBase(const edm::ParameterSet& pset):
+    label(pset.getParameter< std::vector<edm::InputTag> >("label")),
+    bsSrc(pset.getParameter< edm::InputTag >("beamSpot")),
+    label_tp_effic(pset.getParameter< edm::InputTag >("label_tp_effic")),
+    label_tp_fake(pset.getParameter< edm::InputTag >("label_tp_fake")),
+    label_pileupinfo(pset.getParameter< edm::InputTag >("label_pileupinfo")),
+    associators(pset.getParameter< std::vector<std::string> >("associators")),
+    out(pset.getParameter<std::string>("outputFile")),
+    parametersDefiner(pset.getParameter<std::string>("parametersDefiner")),
+    muonHistoParameters(pset.getParameter<edm::ParameterSet>("muonHistoParameters")),
+    ignoremissingtkcollection_(pset.getUntrackedParameter<bool>("ignoremissingtrackcollection",false))
+
+    {
+      dbe_ = edm::Service<DQMStore>().operator->();
```

Comment:
Why do you need to hold a pointer to DQMStore? In the multithread world this is asking for trouble. I saw no real usage (correct) directly of it: I'd suggest to remove it, so the next-to-come developer is not tempted to use it...